processor/configuration/src/test/java/com/redhat/service/smartevents/processor/GatewayConfiguratorImplTest.java

```diff
@@ -0,0 +1,175 @@
+
+import com.redhat.service.smartevents.infra.models.gateways.Action;
+import com.redhat.service.smartevents.infra.models.gateways.Gateway;
+import com.redhat.service.smartevents.infra.models.gateways.Source;
+import com.redhat.service.smartevents.processor.actions.kafkatopic.KafkaTopicAction;
+import com.redhat.service.smartevents.processor.actions.kafkatopic.KafkaTopicActionValidator;
+import com.redhat.service.smartevents.processor.actions.sendtobridge.SendToBridgeAction;
+import com.redhat.service.smartevents.processor.actions.sendtobridge.SendToBridgeActionResolver;
+import com.redhat.service.smartevents.processor.actions.sendtobridge.SendToBridgeActionValidator;
+import com.redhat.service.smartevents.processor.actions.slack.SlackAction;
+import com.redhat.service.smartevents.processor.actions.slack.SlackActionConnector;
+import com.redhat.service.smartevents.processor.actions.slack.SlackActionResolver;
+import com.redhat.service.smartevents.processor.actions.slack.SlackActionValidator;
+import com.redhat.service.smartevents.processor.actions.webhook.WebhookAction;
+import com.redhat.service.smartevents.processor.actions.webhook.WebhookActionValidator;
+
+import io.quarkus.test.junit.QuarkusTest;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@QuarkusTest
+class GatewayConfiguratorImplTest {
+
+    private static final Map<String, ExpectedBeanClasses<Action>> EXPECTED_ACTION_BEANS = Map.of(
+            KafkaTopicAction.TYPE, expect(KafkaTopicActionValidator.class, null, null),
+            SendToBridgeAction.TYPE, expect(SendToBridgeActionValidator.class, SendToBridgeActionResolver.class, null),
+            SlackAction.TYPE, expect(SlackActionValidator.class, SlackActionResolver.class, SlackActionConnector.class),
+            WebhookAction.TYPE, expect(WebhookActionValidator.class, null, null));
+
+    private static final Map<String, ExpectedBeanClasses<Source>> EXPECTED_SOURCE_BEANS = Collections.emptyMap();
+
+    @Inject
+    GatewayConfiguratorImpl configurator;
+
+    @Test
+    void testExpectedActionBeans() {
+        for (Map.Entry<String, ExpectedBeanClasses<Action>> entry : EXPECTED_ACTION_BEANS.entrySet()) {
+            String type = entry.getKey();
+            ExpectedBeanClasses<Action> expected = entry.getValue();
+
+            assertThat(configurator.getActionValidator(type))
+                    .as("GatewayConfigurator.getActionValidator(\"%s\") should not return null", type)
+                    .isNotNull();
+            assertThat(configurator.getActionValidator(type))
+                    .as("GatewayConfigurator.getActionValidator(\"%s\") should be instance of %s", type, expected.validatorClass.getSimpleName())
+                    .isInstanceOf(expected.validatorClass);
+
+            assertThat(configurator.getActionResolver(type))
+                    .as("GatewayConfigurator.getActionResolver(\"%s\") should not return null", type)
+                    .isNotNull();
```

Comment:
Looking on this snippet, I do not understand its importance as in the following if then else block we do basically the same check. Or am I missing something?